torch/csrc/jit/script/tree_views.h

```diff
@@ -794,14 +794,15 @@ struct Const : public Expr {
     tree_->matchNumSubtrees(TK_CONST, 1);
   }
   bool isFloatingPoint() const {
-    return subtree(0)->stringValue().find_first_of(".eE") != std::string::npos;
```

Comment:
Why not just call `strtod` here, similar to how you do for `asIntegral`?

Reply:
If `strtod` can’t do the convention it returns 0, so it can’t determine whether a string is a valid float

Reply:
You can verify whether it succeeded in the conversion by inspecting `endptr`.

Reply:
@driazati is there a reason not to do the suggestion here ?

Reply:
`strtod` succeeds for ints and floats (e.g. `strtod("12")` returns `12.0`)